python/ray/tune/suggest/hebo.py

```diff
@@ -240,17 +239,25 @@ def suggest(self, trial_id: str) -> Optional[Dict]:
                     metric=self._metric,
                     mode=self._mode))
 
+        if not self._live_trial_mapping:
+            self._batch_filled = False
+
         if self._initial_points:
             params = self._initial_points.pop(0)
             suggestion = pd.DataFrame(params, index=[0])
         else:
+            if self._batch_filled or len(
+                    self._live_trial_mapping) >= self._max_concurrent:
+                return None
             if not self._suggestions_cache:
                 suggestion = self._opt.suggest(
-                    n_suggestions=self._n_suggestions)
+                    n_suggestions=self._max_concurrent)
                 self._suggestions_cache = suggestion.to_dict("records")
             params = self._suggestions_cache.pop(0)
             suggestion = pd.DataFrame(params, index=[0])
```

Comment:
If you return n_suggestions here, what happens? Does it return a list of dicts instead?

Reply:
HEBO outputs and inputs dataframes, so with n_suggestions set to eg. 4 it would output a 4 row dataframe